src/App.tsx

```diff
@@ -61,14 +82,24 @@ export const App: React.FC = () => {
       filteredMarkets.map(currentMarket => {
         return getData(`mktDetail?id=${currentMarket.id}`)
           .then(response => checkForError(response))
-          .then(response => response.json());
+          .then(response => response.json())
+          .then(data => cleanDetailsData(data.marketdetails, currentMarket.id))
       })
     )
-      .then(arrayOfPromises => cleanDetailsData(arrayOfPromises))
-      .then(cleanData => setDetails(cleanData))
-      .catch(error => setError(error));
+    .then(data => addScheduleToMarkets(data, filteredMarkets))
+    .then(completeData => setData(completeData))
   };
 
+  const setData = (data: ApiMarkets) => {
+    setMarkets(data.markets)
+    setDetails(data.marketDetails)
+  }
+
+  const findSelectedMarket = (marketID: number) => {
+    setSelectedMarket(marketDetails.find(market => market.id === marketID))
+    history.push(`/markets/${marketID}`);
```

Comment:
Is there a reason to use history.push here? This isn't the best method for moving to a new URL, the only reason I did it for the results page path was because of the button disabling shenanigans that came with using Link (you can't disable a Link). I think we can use a regular ol' route with match.params.id to route to this page.